src/components/Book/Book.tsx

```diff
@@ -40,6 +46,14 @@ const Book = ({
     { id: id },
     { enabled: authStatus === "authenticated" }
   );
+  const responseInvitation = api.user.responseCollaborationInvite.useMutation({
+    onSuccess: () => {
+      void utils.book.invalidate();
+    },
+    onSettled: () => {
+      void utils.book.invalidate();
+    },
+  });
```

Comment:
use either onSuccess or onSettled. This only invalidates the book cache (all book API), perhaps making the website slower. Maybe we need to invalidate the getCollaborationInvite API too. Please recheck this.